### PR TITLE
Add new section on setting up SSH Keys in contributing.md

### LINKED ISSUE #4350: Update Contributing.md to add information about Setting up SSH keys
### Overview
Sometimes new developers run into authorization issues when they attempt to push code to their remote repository. We need to update the [contributing.md](https://github.com/hackforla/website/blob/gh-pages/CONTRIBUTING.md) to include documentation about setting up SSH keys so new developers know how to work past these potential issues.

### Action Items
- [x] Review research in https://github.com/hackforla/website/issues/947
- [x] In the contributing.md, add a new section 2.7.f named "Working on an issue (6): Handling SSH authorization errors"
- [x] Write instructions for setting up SSH kes for both Windows users and Mac users
- [x] View the changes made in your remote repository to ensure the style of the new section is in line with the rest of the contributing.md
- [x] Create a pull request with your changes

### Resources/Instructions
- [Generating a new SSH key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
- [Comment](https://github.com/hackforla/website/issues/947#issuecomment-806257539) with developer research

### PR DESCRIPTION
**Pull request by Jaretzbalba**
Fixes #4350

### What changes did you make and why did you make them ?

  - Add new section 2.7f on how to setup SSH Keys on contributing.md
  - Update table of contents to link to the new section

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

Updating markdown file in website repository. No visual changes to the website.

### CODE REVIEW

**Comment by mjshelton12** [comment]
Review ETA: By end of day 4/8
Availability: Friday 9am-noon, Saturday all day

**Comment by arcan9** [comment]
Review ETA: EOD 04/07
Availability: after 5pm, M-F

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -711,6 +714,79 @@ git checkout update-give-link-2093
...
+2. Then enter the text below into your terminal using your github account email address.
+
+```bash
+ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+```
+
+3. You will be asked to enter a file to save the key, press enter, and then enter a passphrase. Then you will need to ensure that your machine and your repository are connected by running the ssh agent.
+
+```bash
+eval "$(ssh-agent -s)"
```
Comment (arcan9) [inline CONTRIBUTING.md:773]: I've looked at the setup instructions (here: https://github.com/hackforla/website/issues/947#issuecomment-806257539). And it looks like there are actually some steps in the Windows setup that differ slightly from the Mac setup. So, referencing the linked instructions, this line is supposed to be `` eval `ssh-agent -s` ``

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -711,6 +714,79 @@ git checkout update-give-link-2093
...
+```
+
+4. Next you need to add your ssh key to your ssh agent
+
+```bash
+ssh-add ~/.ssh/id_ed25519
```
Comment (arcan9) [inline CONTRIBUTING.md:779]: change to `ssh-add ~/.ssh/id_rsa`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -711,6 +714,79 @@ git checkout update-give-link-2093
...
+```
+
+5. Finally copy the ssh key to add it to your github account by entering the following command. This copies the contents of the id_ed25519.pub file to your clipboard __Don't copy anything else until after you finish the last step, or else you will have to repeat this step.__
+
+```bash
+clip < ~/.ssh/id_ed25519.pub
```
Comment (arcan9) [inline CONTRIBUTING.md:785]: change to `clip < ~/.ssh/id_rsa.pub`

**Review by arcan9 — CHANGES_REQUESTED** [review]
Hi @Jaretzbalba! I just have a few comments addressing some lines of the code. For context, I'm using [this comment](https://github.com/hackforla/website/issues/947#issuecomment-806257539) as reference for the setup instructions, and it looks like there are some discrepancies between what's written in the original comment and in your code. I'm not too familiar with SSH Keys as I've only done this once a long time ago, so feel free clarify some things by replying to my code review, so that I can properly approve your PR!

File: CONTRIBUTING.md (hunk 3 above)
Comment (Jaretzbalba) [reply CONTRIBUTING.md:785]: @arcan9 Yes, I researched into adding SSH Keys using the GitHub documentation and the instructions seemed to differ from what was referenced in the comment for this issue. I figured maybe the info was now outdated or wasn't written correctly so I followed what was currently in the documentation and tested it on my own machine as well. Hope this helps!

**Review by arcan9 — APPROVED** [review]

File: CONTRIBUTING.md (hunk 3 above)
Comment (arcan9) [reply CONTRIBUTING.md:785]: @Jaretzbalba Got it, thanks for clarifying! Approved.

**Comment by mademarc** [comment]
Review ETA: 04/13
Availability: after 7:30pm

**Pull request merged**